### DISCUSSION TITLE
HM600 + OpenDTUonBattery bleibt hängen Bitte um Hilf...

### DISCUSSION THREAD

**Discussion by ozzy19720** [hoylabs/OpenDTU-OnBattery#989] [Q&A] [answered]
Hallo zusammen,
ich brauche bitte dringend noch einmal Eure Hilfe...
Nach dem ich nun umgebaut habe und alles auf LAN gesetzt habe(shelly pro 3em+DTU) wurde es besser, jedoch fäll immer noch hin und wieder meine Regelung aus...
Die DTU gibt eine Regelung raus (gelber Balken), jedoch schaltet der Wechselrichter nicht ein(blauer Balken)
Meisst passiert es, wenn er aus dem Schlaf (0W) in die Regelung gehen soll, aber auch nicht immer...
Das neue Wiki ist klasse geworden und hat mir sehr geholfen, hier komme ich jedoch nicht weiter...
Ich habe es nun geschafft logs mit und ohne diesen Fehler zu loggen und hoffe auf Eure Unterstützung(wieder einmal)
Hier die Links für die Konsolen logs:

https://pastebin.com/XmM5eHRU    Mit Fehler

https://pastebin.com/agM6WvMb    Mit Fehler

https://pastebin.com/Y0FYLxkt          Ohne Fehler

https://pastebin.com/yUJq4nXM       Ohne Fehler

Ich wäre Euch sehr dankbar, wenn Ihr mir hier weiter helfen könntet...
Besten Dank....
MfG
Ozzy

**Comment by ozzy19720** [comment]
P.S. innerhalb der Logs wurden keine Parameter verändert...

**Comment by ozzy19720** [comment]
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164677400/cb8f5981-ac3d-4048-9251-fbc1d01f8b39)

**Comment by ozzy19720** [comment]
Hier bleibt er hängen, angestrebter netzbezug -25w

**Comment by ozzy19720** [comment]
![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164677400/72586275-f88b-46a4-891c-507fdc61e22c)

**Comment by ozzy19720** [comment]
dann fängt er einfach irgendwann wieder an??

**Comment by TomBrett72** [comment, marked as answer]
Hallo :)
hab Deine logfiles mal mit meinen verglichen. Was mir auffällt: das battery inferface steht auf disabled aber es wird nicht ignoriert. Ich hab auch kein SoC, nutze deshalb aktiv die DPL-Spannungsschwellwerte. Könnte mir vorstellen, daß der Inverter sporadisch anspringt, wenn der Solar-Passthrough Leistung liefert.
Deins: [DPL::loop] battery interface disabled, SoC: 0 %, StartTH: 80 %, StopTH: 20 %, SoC age: 1124379 s, ignore: no
Meins: [DPL::loop] battery interface disabled, SoC: 0 %, StartTH: 80 %, StopTH: 20 %, SoC age: 8756 s, ignore: yes

**Reply by ozzy19720** [reply]
Hallo Tom,
vielen Dank für Deine Mühe....und Antwort...
Bin leider kein PC Nerd, sorry...
ist das denn bei den Logs mit und ohne Fehler jeweils gleich bei mir??
Dann hätte es doch nach meinem Verständnis nichts damit zu tun, oder??
Und viel schlimmer....was kann ich dagegen tun???
Besten Dank an Dich...
MfG
Ozzy

**Reply by TomBrett72** [reply]
Keine Ursache, ich bin hier auch nur Nutzer, hätte das niemals programmieren können. Mein HM600 wird auch öfters "gelb". Das kann verschiedene Ursachen haben:
1) Du hast den Luxus einer dicken Batterie mit ~40Volt. Deshalb kannst Du den WR auf vollen 600W laufen lassen - eigentlich. Kann ich mit meinen ~26Volt nicht. Vielleicht könntest Du aber den Test machen und mal auf ~400Watt Spitze runter gehen. Vielleicht ist ja die Leistungsmessung im WR nicht so gut. Wenns tut, stufenweise wieder hoch.
2) Verbindung zum Stromzähler. Wenn die nicht gut ist macht der WR auch gerne mal aus. Kann man testen. Stromzähler abschalten und über die Grundlasteinstellung konstant fahren.
3) Ich dachte bei Dir, daß das Batteriemanagement 0% meldet - also Batterie leer. Dann ja auch WR aus! Hast Du die letzte Version geflasht? Ich finde nämlich den BMS-ignore-Schalter in den DPL Einstellungen nicht mehr. Vielleicht wurde der ins Kapitel Batterie verschoben.
4) schwankende Verbraucher. Bei mir ist die Waschmaschine ein echter Killer für die Regelung. Die Taktung vom Elektroherd ist auch übel.
Alles Punkte, die mich gestern bewogen haben, mich gestern hier anzumelden ;) Das Projekt ist der Hammer, man kann aber prinzipbedingt nicht alles perfektionieren. Es gibt halt beträchtliche Totzeiten, und Strom ist erfordert eigentlich sehr sehr schnelle Regler.
Viel Erfolg!

**Reply by ozzy19720** [reply]
Hey,
habe bist jetzt übergangsweise 36V laufen, die reichen schon aus um auf 600W zu kommen, sollte dann aber mit 48V ausgebaut werden...
mit 24V hast Du recht, da werden es keine 600W... das mit dem runter regeln werde ich gleich mal testen...
Verbindung zum Stromzähler, also zur shelly 3 pro em habe ich schon über Lan Kabel gelöst, ebenso zur DTU...
Jupp, und wie gesagt der soc ignore Schalter wurde zur Batterie gelegt, den siehst Du nur, wenn Du die Batterie Schnittstelle aktivierst...
Bei mir sind es eben nicht die Waschmaschine, Herd und Co.....es tritt einfach sporadisch auf, ohne einen Zusammenhang?!?!
Danke trotzdem....
Und falls Du noch eine Idee hast, gerne her damit.....
Mfg
Ozzy
P.S. Ich will mir keinen Lumentree kaufen....

**Reply by TomBrett72** [reply]
Hi,
hab mir nochmal die logfiles angeschaut. Leider fehlt da die Stelle wo der inverter gestoppt wird. Sieht bei mir so aus wie unten. Das dumme daran, man muß das logging die ganze Zeit mitlaufen lassen, damit man irgendwann die Stelle hat, die den Fehler wirklich auslöst.

19:03:05.067 > [DPL::setNewPowerLimit] input limit: -7 W, min limit: 12 W, max limit: 450 W, hysteresis: 12 W
19:03:05.172 > [DPL::announceStatus] calculated limit is less than minimum power limit
19:03:05.274 > [DPL::updateInverter] Stopping inverter...

MfG!

**Comment by ozzy19720** [comment]
P.S. den soc kann ich nur deaktivieren, wenn ich die Batterieschnittstelle aktiviere, ansonsten ist der im Menu der Batterie im DPL nicht ein zu stellen oder zu sehen...
MfG
Ozzy

**Comment by ozzy19720** [comment]
Hallo zusammen,
hat jemand evtl. noch eine Idee???

MfG
Ozzy

**Reply by spcqike** [reply]
Ich kenne dieses Verhalten bisher nur, wenn das untere Leistungslimit zu gering ist. Du scheinst 15W zu haben, was bei einem HM-600 eigentlich ausreichen sollte. Du kannst es aber gerne mal mit 20W oder mehr probieren. 

Eigentlich gab es mal ein Feature, welches prüft, ob der wechselrichter auch tatsächlich an gegangen ist. Sollte dies nicht passieren, sollte er neugestartet werden (Neustart, nicht einfach nur „start“) 

Hast du die neuste Firmware installiert?

**Reply by ozzy19720** [reply]
Hallo, 
danke für den Tip mit der Mindestleistung...das werde ich noch probieren....
Ich habe an der DTU noch nichts geflasht oder so....habe da immer eine Heidenangst...
Sende ein Bild mit der zur Zeit geflashten Firmenware...
Jupp, und so ein Tool was das ganze überwacht und Neustartet wäre genau das richtige 👍🏻
Wenn ich ihn manuell neustarte, geht es auch wieder.....
Ich habe es einfach nicht hinbekommen das dieser Fehler auftaucht solange ich in der Konsole bin.., manchmal glaube ich, das wenn man in der Weboberfläche der DTU ist, läuft es...
Aber damit kann es doch nichts zu tun haben, oder?
Manchmal läuft er tagelang ohne Probleme und manchmal zig Mal am Tag hängt er sich auf...
Ich finde einfach keinen Zusammenhang....
Mfg
Ozzy
![Screenshot_2024-05-20-19-22-05-280_com android chrome](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164677400/a4932a16-b11b-471e-b475-0e7331687c92)

**Comment by ozzy19720** [comment]
P.S. ist es normal, das der hm600 rot blinkt, wenn der DPL auf null runter regelt, als abschaltet?
Ist mir heute erst aufgefallen???
Mfg
Ozzy

**Reply by TomBrett72** [reply]
Interessant! So schnell wie mein HM600 wieder einschaltet, schaff ichs nie auf den Balkon 🤷🏻‍♂️ Thema flashen: Das eine YouTube hat mich beruhigt, der Mann sagte: man kann beim flashen nichts kaputt machen. Sehr spannend finde ich nur den ersten flash mit *factory. Wenn man dann nicht die richtige PIN-config hat wird es etwas mühselig. Flash OTA lief bis jetzt super 👍 Du hast also auch die generic Version. Wie ich auch. Dachte schon , ich sollte mal die „richtige“ für den ESP32 nehmen. Gestern lief es sogar mit Waschmaschine durch. Heute nur Grundlast und schon 3x AUS. Sehr oft kommt nach einer Abschaltung bei mir „time calibration“ im Inverter Status. Hab deshalb die FRITZBOX zum timeserver gemacht. Scheint aber auch nicht sehr zu helfen… Nunja. spcqike erwähnte ja die Mindeslast. Hatte deshalb schon den Netzbezug auf -30W gestellt. Trotzdem Abschalter, aber mit dem neuesten Flash deutlich weniger. Die Woche gibt’s eh keine Sonne! Kann man in Ruhe flashen 🤓

**Comment by ozzy19720** [comment]
Hallo zusammen,
sorry das ich mich so lange nicht gemeldet habe, war leider beruflich unterwegs...
Die Ausfälle wurden leider nicht besser, egal was ich eingestellt habe, immer das gleiche...
Habe sogar die einzelnen Eingänge des WR einzeln getestet....
Das Einzige wo es evtl. zusammenhängen könnte, das wenn mein 8zusätzliches)BKW einspeist, und ich einen Überschuss habe, also ins Minus komme und ins Netz einspeise, wird es gefühlt schlimmer(ist zusätzlich ein HM600 der aber für sich läuft und nicht in der DTU hinterlegt ist + auf einer anderen Phase als der DPL einspeist)

Habe es heute mal gewagt mir die neueste Firmenware zu flashen und habe alles neu eingestellt.
Hat sogar ganz gut funktioniert...
Bis jetzt läuft es, mal sehen was morgen passiert...
Ich werde berichten...
Besten dank bis jetzt an alle die Geduld mit mir haben und mir Tipps gegeben haben...
MfG
Ozzy

**Comment by ozzy19720** [comment]
Hallo zusammen,
ich habe den ganzen Tag vor der Konsole verbracht und denke ich habe den Fehler in den letzten 3Logs erwischt.
Evtl. kann sich das ja noch einmal jemand ansehen, Danke...

https://pastebin.com/5wyVgGX9  Fehler 3

https://pastebin.com/AZyawugF  Fehler 2

https://pastebin.com/gqCQ6nth  Fehler 1


gefühlt ist es seit dem Update schlimmer geworden...
Was ich bemerkt habe, das es meistens passiert, wenn die Benachrichtigung 15X den "Wechselrichter durch Fernbedienung abgeschaltet " hat, oder zeichnet die DTU nur 15 Benachrichtigungen auf?
Ich weiß mir wirklich keinen Rat mehr...

@spcqike 
Könntest Du mir evtl. einen Rat geben, wo ich dieses "Überwachungstool" finden kann und wie ich es mit einfüge? Ich habe leider nichts gefunden...

Ich schalte das BKW gleich einmal komplett ab, mal sehen was dann passiert..

Vielen Dank für Eure Hilfe
MfG
Ozzy

**Reply by TomBrett72** [reply]
Hi. Es gab noch offene Fragen. 
1) ja, HM blinkt rot wenn über Fernsteuerung abgeschaltet wurde. Hab's ausprobiert.
2) ja, bei 15 Nachrichten ist offenbar der Puffer voll. Ist bei mir auch so.
3) aktuelle Logs. Ich finde immer nur Stellen, bei denen der Stromzähler negative Werte anzeigt. Weshalb soll dann dein BKW einschalten? Hast ein zweites Kraftwerk?! Offensichtlich schiebt dieses genug Strom ins Netz. Oder Du musst die Werte des Stromzählers negieren. Per Definition ist minus immer "zum
Energieversorger zurück".

**Reply by ozzy19720** [reply]
Hey Tom,
DPL hängt auf Phase1, auf 36V System, (zur Zeit nur ein Mppt an Hoymile HM600) , BKW mit HM600 hängt auf Phase 2(aber nicht in DTU eingebunden), wenn das BKW Überschuss produziert, schaltet der DPL natürlich ab, schaltet aber nicht wieder ein, wenn das BKW nicht genug bringt(shelly 3em wieder im Plus ist)
MfG
Ozzy

**Reply by ozzy19720** [reply]
P.S. In der DTU wird es ja erkannt, aber der WR schaltet nicht ein, bis er neu gestartet wird..

**Reply by TomBrett72** [reply]
hast du #809 schon durchgelesen?

**Reply by ozzy19720** [reply]
man,man,man...wo ihr das alles herholt und findet...
ich lese....aber den ersten Zeilen zu folgen ist das ja nicht mein Problem...lese aber weiter...
Bei mir schaltet sich der WR ja nicht ein, nachdem er "eine Zeitlang" abgeschaltet war auf Grund des Überschusses die das BKW erzeugt..
Trotzdem schon einmal danke....
MfG
Orson

**Reply by ozzy19720** [reply]
Bin jetzt auf 50W min. gegangen und auf 100w Grundlast....
ich werde beobachten...
Danke für die Info....
MfG
Ozzy

**Reply by schlimmchen** [reply]
Welche Firmwareversion hast du? In "neueren" wird der WR neugestartet, falls er nicht reagiert auf neue Limits oder auf den Einschaltbefehl.

**Reply by ozzy19720** [reply]
Hallo Bernhard,
habe Donnerstag die letzte aktuelle Version aufgespielt:

![image](https://github.com/helgeerbe/OpenDTU-OnBattery/assets/164677400/34d1f15e-96ae-4aeb-8432-584fc567f2a9)

seit gestern auf 50W min. gegangen, Grundlast 100W, Hys. 10W, Netzbezug -35W max 600W

bis jetzt läuft es, mein BKW ist Richtung Westen, das legt gleich erst los, mal sehen was heute passiert...
Zudem habe ich gestern noch die Batteriekomo. kurz angeschaltet, den SoC auf ignorieren geschaltet, dann Batteriekomo. wieder aus, 
da ich im Log immer gesehen habe das SoC ignorieren auf "no" stand...
Nach dieser Prozedur stand es im Log auf ignorieren "Yes", ob das Ganze damit zu tun hat weiss ich leider nicht, war nur eine Idee...
Ich werde heute Abend noch berichten...

MfG
Ozzy

**Reply by ozzy19720** [reply]
P.S. diese habe ich aufgespielt:
[opendtu-onbattery-generic.bin](https://github.com/helgeerbe/OpenDTU-OnBattery/releases/download/2024.06.03/opendtu-onbattery-generic.bin)
1.86 MB

Wozu ist den diese hier?:

[opendtu-onbattery-generic_esp32.bin](https://github.com/helgeerbe/OpenDTU-OnBattery/releases/download/2024.06.03/opendtu-onbattery-generic_esp32.bin)
1.86 MB


Habe ich da evtl. die falsche drauf??? Ich habe doch ein Esp32, oder sind das die gleichen??

MfG
Ozzy

**Reply by ozzy19720** [reply]
Hallo zusammen,

heute war der erste Tag an dem der WR durchgelaufen ist ohne hängen zu bleiben, oder ich hab es zumindest nicht bemerkt das er neu gestartet hat...
Mit min. 50W scheint es wirklich zu funktionieren, oder wer weiß an was.... evtl. auch das ignorieren des SoC??
Ich beobachte weiter und werde berichten...
Evtl. teste ich bei Gelegenheit auch noch einmal, aber vorerst bin ich froh wie es ist und werde die nächsten Tage sehen ob es stabil bleibt..

Könnte mir evtl. noch jemand helfen zwecks meiner Firmware? Da wäre ich sehr Dankbar für...

Besten Dank...

MfG
Ozzy

**Reply by TomBrett72** [reply]
Hallo @ozzy19720,
Freut mich, sieht also so aus als wäre der SoC Hinweis vom Anfang der Diskussion doch was wert gewesen.
Wegen flash: im originalen openDTU  Projekt gibt es Hinweis dazu: [](https://www.opendtu.solar/firmware/flash_esp/)
Demnach wird es nicht empfohlen, die generic zu benutzen. Updates würde ich aber immer die nutzen, die zum ersten flash passen. Wusste ich vor 3 Monaten auch nicht. Hab deswegen die generic und es tut soweit.

**Comment by schlimmchen** [comment]
> evtl. auch das ignorieren des SoC??

Sehe ich nicht so. Wenn der SoC abgesackt ist und deshalb der WR ausgeschaltet wird, dann wird das aus den Logs auch klar. Außerdem ist laut Logs dein BatteryInterface deaktiviert. Unter diesen Umständen solltest du gar keinen Schalter "SoC ignorieren" sehen können in ein Einstellungen?! Und dann ist es auch egal, wie diese Einstellung ist.

> Mit min. 50W scheint es wirklich zu funktionieren

Ja, das macht Sinn. In deinem "Fehler 3" Log sehe ich, dass der Inverter gestartet werden soll, er das Kommando aber nicht umsetzt. Ich sehe da auch, dass der Counter für aufeinanderfolgende Timeouts beim Versuch den WR zu starten, nicht über 2 kommt, obwohl er das sollte, sodass dann der WR automatisch neu gestartet wird. Das ist wohl ein Fehler. Vermutlich wird der Counter zurückgesetzt, weil das Absetzen des Limits erfolgreich war, obwohl der Start-Befehl wird nicht angenommen wird. Weiter in #1041.

> Könnte mir evtl. noch jemand helfen zwecks meiner Firmware?

Installiere bitte weiter das, was dir bisher als "PIO Umgebung" unter Info -> System angezeigt wird.

Variante `generic` ist veraltet. Es unterscheidet sich von `generic_esp32` nur durch die Definition von Standard-Pins für den NRF24 Transceiver. Das Festlegen der Pins soll aber nicht (mehr) über das PIO Environment passieren, sondern durch die `pin_mapping.json`. Installiere gerne die `generic_esp32`, im Zweifel musst du dann dein Pin Mapping aktualisieren oder sogar überhaupt erst hochladen.
